File: starter_kits/Rust/src/hlt/command.rs
use hlt::direction::Direction;
use hlt::ShipId;

pub struct Command(pub String);

impl Command {
    pub fn spawn_ship() -> Command {
        Command(format!("g"))
    }

    pub fn transform_ship_into_dropoff_site(ship_id: ShipId) -> Command {
        Command(format!("c {}", ship_id.0))
    }

    pub fn move_ship(ship_id: ShipId, direction: Direction) -> Command {
        Command(format!("m {} {}", ship_id.0, direction.get_char_encoding()))
    }
}

File: starter_kits/PHP/hlt/Command.php
<?php

namespace hlt;

class Command
{
    /**
     * @var string
     */
    public $command;

    public static function spawnShip(): Command
    {
        return new Command('g');
    }

    public static function transformShipIntoDropoffSite(EntityId $id): Command
    {
        return new Command('c ' . $id);
    }

    public static function move(EntityId $id, Direction $direction): Command
    {
        return new Command('m ' . $id . ' ' . $direction->getCharValue());
    }

    private function __construct(string $command)
    {
        $this->command = $command;
    }

    public function equals($o): bool
    {
        if (!($o instanceof self)) {
            return false;
        }

        return $this->command === $o->command;
    }
}
